src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -1770,4 +1770,18 @@ public static boolean removeStacksFromProvider(final ICapabilityProvider provide
         return true;
     }
 
+    /**
+     * Create a hard copy of the itemStack list also creating new ItemStack copies.
+     * @param list the list to be copied.
+     * @return the new list.
+     */
+    public static List<ItemStack> hardCopyListOfItemStacks(final List<ItemStack> list)
+    {
+        final List<ItemStack> copy = new ArrayList<>();
```

Comment:
Change this to `new ArrayList<>(list.size());` That way the initial size will be correct and we don't have to worry about any array reallocating.